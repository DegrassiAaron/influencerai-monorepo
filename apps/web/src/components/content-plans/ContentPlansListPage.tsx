'use client';

import Link from 'next/link';
import { useState } from 'react';
import { useQuery } from '@tanstack/react-query';
import { Loader2, Plus, RefreshCw, Search } from 'lucide-react';

import {
  listContentPlans,
  type ContentPlanJob,
  type ListContentPlansParams,
} from '@/lib/content-plans';
import { Button } from '@/components/ui/button';
import { Input } from '@/components/ui/input';
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from '@/components/ui/card';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from '@/components/ui/table';
import { Badge } from '@/components/ui/badge';
import { Alert, AlertDescription, AlertTitle } from '@/components/ui/alert';

const platformLabels: Record<string, string> = {
  instagram: 'Instagram',
  tiktok: 'TikTok',
  youtube: 'YouTube Shorts',
};

function formatDate(value: string) {
  const date = new Date(value);
  if (Number.isNaN(date.getTime())) {
    return '-';
  }
  return new Intl.DateTimeFormat('it-IT', {
    dateStyle: 'medium',
    timeStyle: 'short',
  }).format(date);
}

function ApprovalBadge({ job }: { job: ContentPlanJob }) {
  const status = (job.plan as { approvalStatus?: string }).approvalStatus;
  if (status === 'approved') {
    return (
      <Badge className="bg-emerald-600 text-white hover:bg-emerald-500">Approved</Badge>
    );
  }
  if (status === 'rejected') {
    return (
      <Badge className="bg-destructive text-destructive-foreground hover:bg-destructive/90">
        Rejected
      </Badge>
    );
  }
  return <Badge variant="outline">Pending</Badge>;
}

export function ContentPlansListPage() {
  const [influencerInput, setInfluencerInput] = useState('');
  const [filters, setFilters] = useState<ListContentPlansParams>({});

  const query = useQuery({
    queryKey: ['content-plans', filters],
    queryFn: () => listContentPlans(filters),
  });

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const trimmed = influencerInput.trim();
    setFilters(trimmed ? { influencerId: trimmed } : {});
  };

  const handleResetFilters = () => {
    setInfluencerInput('');
    setFilters({});
  };

  return (
    <div className="space-y-8">
      <header className="flex flex-col gap-4 md:flex-row md:items-center md:justify-between">
        <div className="space-y-1">
          <h1 className="text-3xl font-semibold tracking-tight text-foreground">
            Content Plans
          </h1>
          <p className="text-sm text-muted-foreground">
            Storico dei content plan generati con lo stato di revisione pi√π recente.
          </p>
        </div>
        <div className="flex items-center gap-2">
          <Button
            variant="outline"
            onClick={() => query.refetch()}
            disabled={query.isFetching}
            aria-label="Aggiorna elenco"
          >
            <RefreshCw
              className={`h-4 w-4 ${query.isFetching ? 'animate-spin text-muted-foreground' : ''}`}
            />
          </Button>
          <Button asChild>
            <Link href="/dashboard/content-plans/new">
              <Plus className="mr-2 h-4 w-4" />
              Nuovo content plan
            </Link>
          </Button>
        </div>
      </header>

      <Card>
        <CardHeader>
          <CardTitle>Filtri</CardTitle>
          <CardDescription>Restringi per influencer per ritrovare piani specifici.</CardDescription>
        </CardHeader>
        <CardContent className="space-y-4">
          <form
            onSubmit={handleSubmit}
            className="flex flex-col gap-3 md:flex-row md:items-end"
          >
            <div className="flex-1 space-y-2">
              <label className="text-sm font-medium text-muted-foreground" htmlFor="influencerId">
                Influencer ID
              </label>
              <Input
                id="influencerId"
                placeholder="es. inf_123"
                value={influencerInput}
                onChange={(event) => setInfluencerInput(event.target.value)}
              />
            </div>
            <div className="flex gap-2">
              <Button type="submit" className="gap-2">
                <Search className="h-4 w-4" />
                Applica
              </Button>
              <Button
                type="button"
                variant="ghost"
                onClick={handleResetFilters}
                disabled={!influencerInput && Object.keys(filters).length === 0}
              >
                Reset
              </Button>
            </div>
          </form>

          {query.isError ? (
            <Alert variant="destructive">
              <AlertTitle>Errore durante il caricamento</AlertTitle>
              <AlertDescription>
                {(query.error as Error).message ?? 'Impossibile recuperare i content plan.'}
              </AlertDescription>
            </Alert>
          ) : null}

          {query.isPending ? (
            <div className="flex justify-center py-10">
              <Loader2 className="h-6 w-6 animate-spin text-muted-foreground" aria-label="Loading" />
            </div>
          ) : null}

          {query.isSuccess && query.data.length === 0 ? (
            <div className="flex flex-col items-center justify-center gap-3 rounded-lg border border-dashed border-muted-foreground/40 py-12 text-center">
              <p className="text-sm text-muted-foreground">
                Nessun content plan trovato con i filtri correnti.
              </p>
              <Button asChild>
                <Link href="/dashboard/content-plans/new">
                  <Plus className="mr-2 h-4 w-4" />
                  Genera il primo piano
                </Link>
              </Button>
            </div>
          ) : null}

          {query.isSuccess && query.data.length > 0 ? (
            <div className="rounded-md border border-border">
              <Table>
                <TableHeader>
                  <TableRow>
                    <TableHead>Creato</TableHead>
                    <TableHead>Influencer</TableHead>
                    <TableHead>Tema</TableHead>
                    <TableHead>Piattaforme</TableHead>
                    <TableHead className="text-center">Post</TableHead>
                    <TableHead className="text-center">Stato</TableHead>
                    <TableHead>Job ID</TableHead>
                  </TableRow>
                </TableHeader>
                <TableBody>
                  {query.data.map((job) => (
                    <TableRow key={job.id}>
                      <TableCell>{formatDate(job.plan.createdAt)}</TableCell>
                      <TableCell>
                        <span className="rounded bg-muted px-2 py-1 text-xs font-medium">
                          {job.plan.influencerId}
                        </span>
                      </TableCell>
                      <TableCell className="max-w-xs truncate">{job.plan.theme}</TableCell>
                      <TableCell>
                        <div className="flex flex-wrap gap-2">
                          {job.plan.targetPlatforms.map((platform) => (
                            <Badge key={`${job.id}-${platform}`} variant="secondary">
                              {platformLabels[platform] ?? platform}
                            </Badge>
                          ))}
                        </div>
                      </TableCell>
                      <TableCell className="text-center">{job.plan.posts.length}</TableCell>
                      <TableCell className="text-center">
                        <ApprovalBadge job={job} />
                      </TableCell>
                      <TableCell>
                        <code className="text-xs text-muted-foreground">{job.id}</code>
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </div>
          ) : null}
        </CardContent>
      </Card>
    </div>
  );
}
