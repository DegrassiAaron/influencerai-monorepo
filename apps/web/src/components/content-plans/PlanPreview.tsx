import React from 'react';

import { Card, CardContent, CardHeader, CardTitle } from '../ui/card';
import { Badge } from '../ui/badge';
import { Button } from '../ui/button';
import type { ContentPlanJob } from '../../lib/content-plans';

export type PlanPreviewProps = {
  job: ContentPlanJob;
  approvalStatus: 'approved' | 'rejected' | null;
  onRegenerate: () => void;
  onApprove: () => void;
  onReject: () => void;
  isRegenerating: boolean;
  isUpdatingApproval: boolean;
};

const platformLabels: Record<string, string> = {
  instagram: 'Instagram',
  tiktok: 'TikTok',
  youtube: 'YouTube Shorts',
};

export function PlanPreview({
  job,
  approvalStatus,
  onRegenerate,
  onApprove,
  onReject,
  isRegenerating,
  isUpdatingApproval,
}: PlanPreviewProps) {
  return (
    <section className="space-y-4" aria-live="polite">
      <header className="flex flex-col gap-2 md:flex-row md:items-center md:justify-between">
        <div>
          <h2 className="text-xl font-semibold text-foreground">Generated content plan</h2>
          <p className="text-sm text-muted-foreground">
            Piano creato il {new Date(job.plan.createdAt).toLocaleString()}. Usa le azioni per
            approvare o richiedere una nuova generazione.
          </p>
        </div>
        <div className="flex flex-wrap gap-2">
          <Button variant="outline" onClick={onRegenerate} disabled={isRegenerating}>
            {isRegenerating ? 'Rigenerazione...' : 'Regenerate plan'}
          </Button>
          <Button
            variant="default"
            className="bg-emerald-600 text-white hover:bg-emerald-500"
            onClick={onApprove}
            disabled={isUpdatingApproval}
          >
            {isUpdatingApproval && approvalStatus !== 'approved'
              ? 'Salvataggio...'
              : 'Approve plan'}
          </Button>
          <Button variant="destructive" onClick={onReject} disabled={isUpdatingApproval}>
            {isUpdatingApproval && approvalStatus !== 'rejected' ? 'Salvataggio...' : 'Reject plan'}
          </Button>
        </div>
      </header>
      {approvalStatus ? (
        <p className="text-sm font-medium text-muted-foreground">
          Stato corrente: <span className="capitalize text-foreground">{approvalStatus}</span>
        </p>
      ) : null}
      <div className="grid gap-4 md:grid-cols-2">
        {job.plan.posts.map((post, index) => (
          <Card key={`${job.id}-${index}`}>
            <CardHeader>
              <CardTitle className="text-base font-semibold">Post {index + 1}</CardTitle>
              <p className="text-xs text-muted-foreground">
                Piattaforme:{' '}
                {job.plan.targetPlatforms.map((p) => platformLabels[p] ?? p).join(', ')}
              </p>
            </CardHeader>
            <CardContent className="space-y-3">
              <div>
                <h3 className="text-sm font-medium text-muted-foreground">Caption</h3>
                <p className="mt-1 text-sm leading-relaxed text-foreground">{post.caption}</p>
              </div>
              <div>
                <h3 className="text-sm font-medium text-muted-foreground">Hashtags</h3>
                <div className="mt-2 flex flex-wrap gap-2">
                  {post.hashtags.map((tag) => (
                    <Badge key={tag} variant="secondary">
                      {tag}
                    </Badge>
                  ))}
                </div>
              </div>
            </CardContent>
          </Card>
        ))}
      </div>
    </section>
  );
}
