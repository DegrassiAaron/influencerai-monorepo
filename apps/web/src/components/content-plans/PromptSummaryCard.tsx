import React from 'react';

import { Card, CardContent, CardHeader, CardTitle } from '../ui/card';
import { Badge } from '../ui/badge';

export type PromptSummaryCardProps = {
  personaSummary: string;
  theme: string;
  targetPlatforms: string[];
};

const platformLabels: Record<string, string> = {
  instagram: 'Instagram',
  tiktok: 'TikTok',
  youtube: 'YouTube Shorts',
};

export function PromptSummaryCard({
  personaSummary,
  theme,
  targetPlatforms,
}: PromptSummaryCardProps) {
  return (
    <Card data-testid="prompt-summary">
      <CardHeader>
        <CardTitle className="text-xl font-semibold">Review prompt configuration</CardTitle>
        <p className="text-sm text-muted-foreground">
          Conferma i parametri che verranno inviati alla generazione del piano editoriale.
        </p>
      </CardHeader>
      <CardContent className="space-y-4">
        <section>
          <h3 className="text-sm font-medium text-muted-foreground">Persona</h3>
          <p className="mt-1 text-sm leading-relaxed text-foreground">{personaSummary}</p>
        </section>
        <section>
          <h3 className="text-sm font-medium text-muted-foreground">Theme</h3>
          <p className="mt-1 text-sm text-foreground">{theme}</p>
        </section>
        <section>
          <h3 className="text-sm font-medium text-muted-foreground">Target platforms</h3>
          <div className="mt-2 flex flex-wrap gap-2">
            {targetPlatforms.map((platform) => (
              <Badge key={platform} variant="outline">
                {platformLabels[platform] ?? platform}
              </Badge>
            ))}
          </div>
        </section>
      </CardContent>
    </Card>
  );
}
