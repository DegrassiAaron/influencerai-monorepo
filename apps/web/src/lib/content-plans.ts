import { ContentPlanSchema } from '@influencerai/core-schemas';
import { z } from 'zod';

import { apiGet, apiPatch, apiPost } from './api';

const PlatformEnum = z.enum(['instagram', 'tiktok', 'youtube']);

const CreateContentPlanInputSchema = z.object({
  influencerId: z.string().min(1, 'Influencer ID is required'),
  theme: z.string().min(1, 'Theme is required'),
  targetPlatforms: z.array(PlatformEnum).min(1, 'Select at least one platform'),
});

const ApprovalStatusSchema = z.enum(['approved', 'rejected']);

const ContentPlanWithApprovalSchema = ContentPlanSchema.extend({
  approvalStatus: ApprovalStatusSchema.optional(),
});

const ContentPlanJobSchema = z.object({
  id: z.string(),
  plan: ContentPlanWithApprovalSchema,
});

const UpdateContentPlanApprovalSchema = z.object({
  id: z.string().min(1),
  plan: ContentPlanWithApprovalSchema,
  approvalStatus: ApprovalStatusSchema,
});

const ListContentPlansParamsSchema = z.object({
  influencerId: z.string().min(1).optional(),
  take: z.number().int().min(1).max(100).optional(),
  skip: z.number().int().min(0).optional(),
});

const ContentPlanJobListSchema = z.array(ContentPlanJobSchema);

export type CreateContentPlanInput = z.infer<typeof CreateContentPlanInputSchema>;
export type ApprovalStatus = z.infer<typeof ApprovalStatusSchema>;
export type ContentPlanJob = z.infer<typeof ContentPlanJobSchema>;
export type UpdateContentPlanApprovalInput = z.infer<typeof UpdateContentPlanApprovalSchema>;
export type ListContentPlansParams = z.infer<typeof ListContentPlansParamsSchema>;

export async function createContentPlan(input: CreateContentPlanInput): Promise<ContentPlanJob> {
  const payload = CreateContentPlanInputSchema.parse(input);
  const response = await apiPost<CreateContentPlanInput, unknown>('/content-plans', payload);
  const parsed = ContentPlanJobSchema.safeParse(response);
  if (!parsed.success) {
    throw new Error('Invalid response format from content plan API');
  }
  return parsed.data;
}

export async function updateContentPlanApproval(input: UpdateContentPlanApprovalInput) {
  const payload = UpdateContentPlanApprovalSchema.parse(input);
  const planWithStatus = { ...payload.plan, approvalStatus: payload.approvalStatus };
  await apiPatch(`/jobs/${payload.id}`, { result: planWithStatus });
  return { ...payload, plan: planWithStatus };
}

export async function listContentPlans(params: ListContentPlansParams = {}) {
  const parsed = ListContentPlansParamsSchema.parse(params ?? {});
  const searchParams = new URLSearchParams();
  if (parsed.influencerId) {
    searchParams.set('influencerId', parsed.influencerId);
  }
  if (typeof parsed.take === 'number') {
    searchParams.set('take', String(parsed.take));
  }
  if (typeof parsed.skip === 'number') {
    searchParams.set('skip', String(parsed.skip));
  }
  const queryString = searchParams.toString();
  const path = queryString ? `/content-plans?${queryString}` : '/content-plans';
  const response = await apiGet<unknown>(path);
  const parsedResponse = ContentPlanJobListSchema.safeParse(response);
  if (!parsedResponse.success) {
    throw new Error('Invalid response format while listing content plans');
  }
  return parsedResponse.data;
}
