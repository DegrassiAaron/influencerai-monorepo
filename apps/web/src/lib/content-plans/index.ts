export * from './types';
export * from './prompt';
export * from './api';
