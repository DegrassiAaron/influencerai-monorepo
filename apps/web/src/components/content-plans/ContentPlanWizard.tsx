'use client';

import React, { useMemo, useState } from 'react';
import { useMutation } from '@tanstack/react-query';

import { Button } from '../ui/button';
import { Input } from '../ui/input';
import { Label } from '../ui/label';
import { Badge } from '../ui/badge';
import { Textarea } from '../ui/textarea';
import { cn } from '../../lib/utils';
import {
  createContentPlan,
  updateContentPlanApproval,
  type ApprovalStatus,
  type ContentPlanJob,
} from '../../lib/content-plans';

import { PromptSummaryCard } from './PromptSummaryCard';
import { PlanPreview } from './PlanPreview';

type WizardStep = 0 | 1 | 2;

type WizardState = {
  influencerId: string;
  personaSummary: string;
  theme: string;
  targetPlatforms: Platform[];
};

type Platform = 'instagram' | 'tiktok' | 'youtube';

const platformOptions: Array<{ value: Platform; label: string; description: string }> = [
  {
    value: 'instagram',
    label: 'Instagram',
    description: 'Stories e reel brevi per mantenere ingaggio costante.',
  },
  {
    value: 'tiktok',
    label: 'TikTok',
    description: 'Short form video con trend del momento.',
  },
  {
    value: 'youtube',
    label: 'YouTube Shorts',
    description: 'Clip verticali ottimizzate per discovery.',
  },
];

const steps: Array<{ title: string; description: string }> = [
  {
    title: 'Persona',
    description: "Identifica l'influencer virtuale e sintetizza i tratti chiave.",
  },
  {
    title: 'Parametri',
    description: 'Definisci tema editoriale e piattaforme prioritarie.',
  },
  {
    title: 'Revisione',
    description: 'Controlla il prompt e avvia la generazione del piano.',
  },
];

const initialState: WizardState = {
  influencerId: '',
  personaSummary: '',
  theme: '',
  targetPlatforms: ['instagram'],
};

export function ContentPlanWizard() {
  const [step, setStep] = useState<WizardStep>(0);
  const [state, setState] = useState<WizardState>(initialState);
  const [job, setJob] = useState<ContentPlanJob | null>(null);
  const [approvalStatus, setApprovalStatus] = useState<ApprovalStatus | null>(null);

  const createPlanMutation = useMutation({
    mutationFn: createContentPlan,
    onSuccess: (result) => {
      setJob(result);
      setApprovalStatus(null);
    },
  });

  const approvalMutation = useMutation({
    mutationFn: updateContentPlanApproval,
    onSuccess: (data) => {
      setApprovalStatus(data.approvalStatus);
      setJob((prev) => (prev ? { ...prev, plan: data.plan } : prev));
    },
  });

  const canGoNext = useMemo(() => {
    if (step === 0) {
      return state.influencerId.trim().length > 0 && state.personaSummary.trim().length > 0;
    }
    if (step === 1) {
      return state.theme.trim().length > 0 && state.targetPlatforms.length > 0;
    }
    return true;
  }, [state.influencerId, state.personaSummary, state.theme, state.targetPlatforms, step]);

  const handleNext = () => {
    if (step < 2 && canGoNext) {
      setStep((prev) => (prev + 1) as WizardStep);
    }
  };

  const handleBack = () => {
    if (step > 0) {
      setStep((prev) => (prev - 1) as WizardStep);
    }
  };

  const resetGeneratedPlan = () => {
    if (job) {
      setJob(null);
    }
    setApprovalStatus(null);
  };

  const setField = <K extends keyof WizardState>(key: K, value: WizardState[K]) => {
    setState((prev) => ({ ...prev, [key]: value }));
    resetGeneratedPlan();
  };

  const togglePlatform = (platform: Platform) => {
    setState((prev) => {
      const exists = prev.targetPlatforms.includes(platform);
      const targetPlatforms = exists
        ? prev.targetPlatforms.filter((p) => p !== platform)
        : [...prev.targetPlatforms, platform];
      return { ...prev, targetPlatforms };
    });
    resetGeneratedPlan();
  };

  const handleGenerate = () => {
    createPlanMutation.mutate({
      influencerId: state.influencerId.trim(),
      theme: state.theme.trim(),
      targetPlatforms: state.targetPlatforms,
    });
  };

  const handleApproval = (status: ApprovalStatus) => {
    if (!job) return;
    approvalMutation.mutate({
      id: job.id,
      approvalStatus: status,
      plan: {
        ...job.plan,
        posts: job.plan.posts,
      },
    });
  };

  const resetWizard = () => {
    setState(initialState);
    setStep(0);
    setJob(null);
    setApprovalStatus(null);
  };

  return (
    <div className="mx-auto flex w-full max-w-5xl flex-col gap-8">
      <header className="space-y-4">
        <h1 className="text-3xl font-semibold text-foreground">Content plan wizard</h1>
        <p className="max-w-3xl text-sm text-muted-foreground">
          Guida marketing step-by-step per configurare i prompt, generare anteprime dei post e
          condividere decisioni di approvazione con il team operativo.
        </p>
        <ol className="flex flex-col gap-3 md:flex-row md:gap-4">
          {steps.map((item, index) => (
            <li
              key={item.title}
              className={cn(
                'flex-1 rounded-lg border p-3',
                index === step ? 'border-brand-500 bg-brand-50' : 'border-border'
              )}
            >
              <div className="flex items-center gap-2">
                <Badge variant={index === step ? 'brand' : 'outline'}>{index + 1}</Badge>
                <span className="text-sm font-medium text-foreground">{item.title}</span>
              </div>
              <p className="mt-1 text-xs text-muted-foreground">{item.description}</p>
            </li>
          ))}
        </ol>
      </header>

      <section className="rounded-xl border border-border bg-card p-6 shadow-sm">
        {step === 0 ? (
          <div className="space-y-6">
            <div className="space-y-2">
              <Label htmlFor="influencerId">Influencer ID</Label>
              <Input
                id="influencerId"
                placeholder="es. inf_1234"
                value={state.influencerId}
                onChange={(event) => setField('influencerId', event.target.value)}
              />
              <p className="text-xs text-muted-foreground">
                Inserisci l&apos;identificativo dell&apos;influencer virtuale da utilizzare
                nell&apos;API.
              </p>
            </div>
            <div className="space-y-2">
              <Label htmlFor="personaSummary">Persona summary</Label>
              <Textarea
                id="personaSummary"
                placeholder="Descrivi tono di voce, interessi e community dell'influencer virtuale"
                value={state.personaSummary}
                onChange={(event) => setField('personaSummary', event.target.value)}
              />
              <p className="text-xs text-muted-foreground">
                Questa descrizione aiuta il team a contestualizzare il piano generato e resta
                disponibile nella revisione finale.
              </p>
            </div>
          </div>
        ) : null}

        {step === 1 ? (
          <div className="space-y-6">
            <div className="space-y-2">
              <Label htmlFor="theme">Theme</Label>
              <Input
                id="theme"
                placeholder="es. Lancio programma Summer Glow"
                value={state.theme}
                onChange={(event) => setField('theme', event.target.value)}
              />
            </div>
            <div className="space-y-3">
              <span className="text-sm font-medium text-foreground">Target platforms</span>
              <div className="grid gap-3 md:grid-cols-3">
                {platformOptions.map((option) => {
                  const checked = state.targetPlatforms.includes(option.value);
                  return (
                    <button
                      key={option.value}
                      type="button"
                      onClick={() => togglePlatform(option.value)}
                      className={cn(
                        'rounded-lg border p-3 text-left transition',
                        checked
                          ? 'border-brand-500 bg-brand-50'
                          : 'border-border hover:border-brand-300'
                      )}
                    >
                      <div className="flex items-center justify-between">
                        <span className="text-sm font-semibold text-foreground">
                          {option.label}
                        </span>
                        {checked ? <Badge variant="brand">Selected</Badge> : null}
                      </div>
                      <p className="mt-2 text-xs leading-relaxed text-muted-foreground">
                        {option.description}
                      </p>
                    </button>
                  );
                })}
              </div>
            </div>
          </div>
        ) : null}

        {step === 2 ? (
          <div className="space-y-6">
            <PromptSummaryCard
              personaSummary={state.personaSummary}
              theme={state.theme}
              targetPlatforms={state.targetPlatforms}
            />
            {createPlanMutation.isError ? (
              <p className="text-sm text-destructive">
                {(createPlanMutation.error as Error)?.message ??
                  'Errore durante la generazione del piano.'}
              </p>
            ) : null}
            <div className="flex flex-wrap gap-3">
              <Button onClick={handleGenerate} disabled={createPlanMutation.isPending}>
                {createPlanMutation.isPending
                  ? 'Generazione...'
                  : job
                    ? 'Regenerate content plan'
                    : 'Generate content plan'}
              </Button>
              <Button variant="ghost" onClick={resetWizard}>
                Reset form
              </Button>
            </div>
            {job ? (
              <PlanPreview
                job={job}
                approvalStatus={approvalStatus}
                onRegenerate={handleGenerate}
                onApprove={() => handleApproval('approved')}
                onReject={() => handleApproval('rejected')}
                isRegenerating={createPlanMutation.isPending}
                isUpdatingApproval={approvalMutation.isPending}
              />
            ) : null}
            {approvalMutation.isError ? (
              <p className="text-sm text-destructive">
                {(approvalMutation.error as Error)?.message ??
                  'Impossibile aggiornare lo stato di approvazione.'}
              </p>
            ) : null}
          </div>
        ) : null}
      </section>

      <footer className="flex items-center justify-between">
        <Button variant="outline" onClick={handleBack} disabled={step === 0}>
          Back
        </Button>
        {step < 2 ? (
          <Button onClick={handleNext} disabled={!canGoNext}>
            Next
          </Button>
        ) : (
          <div className="h-10" aria-hidden="true" />
        )}
      </footer>
    </div>
  );
}
