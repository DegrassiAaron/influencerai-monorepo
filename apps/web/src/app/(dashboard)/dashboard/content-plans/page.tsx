import { Metadata } from 'next';

import { ContentPlanWizard } from '@/components/content-plans/ContentPlanWizard';

export const metadata: Metadata = {
  title: 'Content Plans Wizard',
  description: 'Configura, genera e approva piani editoriali guidati.',
};

export default function ContentPlansPage() {
  return <ContentPlanWizard />;
}
