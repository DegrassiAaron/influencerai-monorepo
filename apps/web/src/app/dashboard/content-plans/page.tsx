import { ContentPlanWizard } from "@/components/content-plans/ContentPlanWizard";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";

export default function ContentPlansPage() {
  return (
    <div className="flex flex-col gap-6">
      <header className="flex flex-col gap-4 border-b border-border/60 pb-6 md:flex-row md:items-center md:justify-between">
        <div className="space-y-2">
          <Badge variant="brand" className="bg-brand-100 text-brand-700">
            Content Planning
          </Badge>
          <h1 className="text-3xl font-semibold text-foreground">
            Genera e approva il piano editoriale
          </h1>
          <p className="max-w-2xl text-sm text-muted-foreground">
            Completa il wizard per creare un piano editoriale personalizzato per
            la tua persona e gestisci l'approvazione del contenuto.
          </p>
        </div>
        <Button variant="secondary">Scarica come CSV</Button>
      </header>
      <ContentPlanWizard />
    </div>
  );
}
