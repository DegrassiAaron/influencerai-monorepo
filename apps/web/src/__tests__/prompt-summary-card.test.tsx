import React from "react";
import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

import { PromptSummaryCard } from "../components/content-plans/PromptSummaryCard";

describe("PromptSummaryCard", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <PromptSummaryCard
        personaSummary="Energetic AI influencer with a focus on wellness"
        theme="Summer Glow"
        targetPlatforms={["instagram", "tiktok"]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
