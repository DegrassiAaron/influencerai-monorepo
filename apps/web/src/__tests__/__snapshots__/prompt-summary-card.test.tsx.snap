// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PromptSummaryCard > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl border border-border bg-card text-card-foreground shadow-sm"
    data-testid="prompt-summary"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="tracking-tight text-xl font-semibold"
      >
        Review prompt configuration
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        Conferma i parametri che verranno inviati alla generazione del piano editoriale.
      </p>
    </div>
    <div
      class="p-6 pt-0 space-y-4"
    >
      <section>
        <h3
          class="text-sm font-medium text-muted-foreground"
        >
          Persona
        </h3>
        <p
          class="mt-1 text-sm leading-relaxed text-foreground"
        >
          Energetic AI influencer with a focus on wellness
        </p>
      </section>
      <section>
        <h3
          class="text-sm font-medium text-muted-foreground"
        >
          Theme
        </h3>
        <p
          class="mt-1 text-sm text-foreground"
        >
          Summer Glow
        </p>
      </section>
      <section>
        <h3
          class="text-sm font-medium text-muted-foreground"
        >
          Target platforms
        </h3>
        <div
          class="mt-2 flex flex-wrap gap-2"
        >
          <div
            class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors text-foreground"
          >
            Instagram
          </div>
          <div
            class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors text-foreground"
          >
            TikTok
          </div>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
